R/add_gnomAD_AF.R

```diff
@@ -85,3 +64,64 @@ add_gnomAD_AF <- function(data,
   mafdb
 }
 
+#' @rdname add_gnomAD_AF
+#'
+#' @examples
+#' \dontrun{
+#' BiocManager::install("MafDb.gnomAD.r2.1.hs37d5")
+#' file <- system.file("extdata", "allelic_counts_HG00187.csv", package = "tMAE", mustWork = TRUE)
+#' maeCounts <- fread(file)
+#' maeRes <- DESeq4MAE(maeCounts)
+#' genome_assembly <- 'hg19'
+#' res <- add_gnomAD_AF(maeRes, genome_assembly = genome_assembly, pop="AF")
+#' }
+#'
+setMethod("add_gnomAD_AF", signature = "data.table", 
+function(
+    object, 
+    genome_assembly = 'hg19',
+    max_af_cutoff = .001,
+    populations = c('AF', 'AF_afr', 'AF_amr', 'AF_eas', 'AF_nfe', 'AF_popmax'),
+    ... ){
+  # create GRanges from a MAE count table
+  gr <- GRanges(seqnames = object$contig,
+                ranges = IRanges(start=object$position, width=1), 
+                strand = '*')
+  # score the gr
```

Comment:
rather call here `add_gnomad_AF` with your GRange object. This way you have the same logic for all input data and don't duplicate code.